### DISCUSSION TITLE
cant download videos

### DISCUSSION THREAD

**Discussion by franxd2** [aclap-dev/video-downloadhelper#309] [Help and Q&A]
### Your question

i try to download videos and i get this

"Not detecting m3u8/hls with non standard extension and non standard mime type [mp4 @ 00000233274a0100] Could not find tag for codec ansi in stream #0, codec not currently supported in container [out#0/mp4 @ 0000023325806b80] Could not write header (incorrect codec parameters ?): Invalid argument Error opening output file E:\NUEVOS\filename.mp4. Error opening output files: Invalid argument "

can some help me
thx

### Addon & CoApp version and details

version: 8.2.2.8
channel: stable
build date: 2024-03-18
build options: linuxlic: false, noyt: true, browser: chrome
lang: es-ES
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\fran",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by jcfrog** [comment]
hi,
can you share a public link for a video with this issue?

**Reply by franxd2** [reply]
sure!
https://hentaila.tv/ver/modaete-yo-adam-kun/episodio-7/

**Reply by GrampaWildWilly** [reply]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3296, Firefox 123.0.1 64-bit, licensed VDH 8.2.2.8a1 beta, CoApp 2.0.19.  I do not work for VDH.  (@jcfrog Jérôme does.)  I am just another user.

I tried to download the video on that page & also got the same error you did.  Time for Plan B.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cc57d8f3-8764-4cf6-902f-6db6f24c7696)

Running that URL through ffprobe got me this:

[ffprobe.txt](https://github.com/aclap-dev/video-downloadhelper/files/14641971/ffprobe.txt)

Using that information, I downloaded the item using ffmpeg:

[Hentai mp4 Log.txt](https://github.com/aclap-dev/video-downloadhelper/files/14641976/Hentai.mp4.Log.txt)

The resulting file looked like this:

![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/cb0edf1e-539d-4cfd-8931-f27afeafd3ca)

I did manage to find a part of this video that was not X rated:

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/c5f21053-f8fe-4fcc-8a37-1b14ee6bf888)

I hope the Japanese writing in there is not X rated.  In any case, I did not sit & watch this.  It is cartoon sex.  That's what Hentai is.  I find that boring.  In any case, I just sampled this at intervals & it was fine, video & audio at each sample point, all the way to the end.

I don't understand what tripped up VDH here.  This HLS master manifest is pretty ordinary.  Maybe the fact that the file names of the stream manifests & the file names of the individual chunks aren't as expected.  The stream manifests are not named with the file extension .m3u8, which would be standard.  They are .txt, which is unusual.  Also the chunks have file names that don't even have an extension.  What I usually see is that they are .ts files.  Despite these deviations from the standard, ffmpeg had no problem downloading this.  The CoApp uses ffmpeg.  In fact, if you look closely, you'll see that I used the CoApp's ffmpeg to download this.  This is rather baffling.

**Comment by jcfrog** [comment]
@paulrouget can you have a look?
